File: python/shared/dtos/position_dto.py
from . import BaseDTO, NameStr, NullableUrlStr


class PositionDTO(BaseDTO):
    position_id: int
    preset_id: int
    name: str
    icon_url: str | None

    model_config = {"from_attributes": True}


class AddPositionDTO(BaseDTO):
    name: NameStr
    icon_url: NullableUrlStr


class UpdatePositionDTO(BaseDTO):
    name: NameStr | None = None
    icon_url: NullableUrlStr = None

File: web/src/dtos/positionDto.ts
export interface PositionDTO {
  positionId: number;
  presetId: number;
  name: string;
  iconUrl?: string | null;
}

export interface AddPositionDTO {
  name: string;
  iconUrl: string | null;
}

export interface UpdatePositionDTO {
  name?: string;
  iconUrl?: string | null;
}
